Tools/cases_generator/stack.py

```diff
@@ -125,7 +125,7 @@ def __init__(self) -> None:
         self.variables: list[StackItem] = []
         self.defined: set[str] = set()
 
-    def pop(self, var: StackItem) -> str:
+    def pop(self, var: StackItem, is_abstract: bool = False) -> str:
```

Comment:
I don't like leaking details of the code generator into the low level components.

What fails if this change is not made?

Reply:
The optimizer generator tries to generate `.bits = xyz` but it doesn't need it because it's using the `_PyUOpSymbol *` not `_PyStackRef`

Reply:
Yes, but what fails?
Do you have an example?

Reply:
Yes the `_PUSH_FRAME` and friends, because those cast to `_PyInterpreterFrame *` which causes a commpilatione rror.

Reply:
The `is_abstract` flag requires implicit knowledge of how the abstract code generator works and couples `stack.py` to the abstract interpreter.

Pass in something that makes sense to at the lower level, e.g. a pattern for casting like "{item}.bits" for the executors and "{type}{item}" for the abstract interpreter.